src/lookup/derive.ts

```diff
@@ -60,56 +115,20 @@ export default async function getVerifiableCredentialAllFromShape(
   if (internalOptions.fetch === undefined) {
     internalOptions.fetch = fallbackFetch;
   }
-  // credentialClaims should contain all the claims, but not the context.
-  // const { "@context": claimsContext, ...credentialClaims } = vcShape;
-  // The following lines refactor the previous deconstruction in order to work
-  // around a misalignment between `rollup-plugin-typescript2` and NodeJS.
-  // Issue tracking: https://github.com/ezolenko/rollup-plugin-typescript2/issues/282
-  const credentialClaims = { ...vcShape };
-  delete credentialClaims["@context"];
-  const claimsContext = vcShape["@context"];
-  const credentialRequestBody: {
-    verifiableCredential: Partial<VerifiableCredential>;
-    options?: { include: typeof INCLUDE_EXPIRED_VC_OPTION };
-  } = {
-    // See https://w3c-ccg.github.io/vc-api/holder.html
-    verifiableCredential: {
-      "@context": concatenateContexts(defaultContext, claimsContext),
-      ...credentialClaims,
-    },
-  };
-  if (internalOptions.includeExpiredVc) {
-    credentialRequestBody.options = {
-      include: INCLUDE_EXPIRED_VC_OPTION,
-    };
-  }
-  const response = await internalOptions.fetch(holderEndpoint, {
-    headers: {
-      "Content-Type": "application/json",
-    },
-    method: "POST",
-    body: JSON.stringify(credentialRequestBody),
+  // The request payload depends on the target endpoint.
+  const vpRequest = holderEndpoint.endsWith("/query")
```

Comment:
It might be an idea to use legacy only if it's `/derive` here, and default to trying `/query` otherwise?

Reply:
It was my initial approach, but the legacy endpoints come with a discovery mechanism, so it could not be `/derive` and still be valid, while the `/query` endpoint is a convention. That's why I added a comment on the following lines.